### DISCUSSION TITLE
default_cell_formatter() in DataTable formats float to 2-digit scale, which is often undesirable.

### DISCUSSION THREAD

**Discussion by grvstick** [Textualize/textual#4993] [Ideas]
#### Code

https://github.com/Textualize/textual/blob/db3fda928ba3f4807f01ae24ecee260ce6206b1b/src/textual/widgets/_data_table.py#L197

This defaults all floats to be in 2-digit scale. I think it would be better to change this to simple float-to-str conversion, so that the user can decide the scale. If there is a specific reason to limit the float type scales, the scale should be left to maximum to the limits.

**Comment by darrenburns** [comment]
@grvstick It likely was a virus.

The original plan was to expose this function to end users, but that never came to fruition. Hopefully DataTable can get a bit of love soon to give more control over things like this 😄

**Comment by brett-talko** [comment]
Just came to this discussion because I noticed the same thing. Ideally we'd like to specify the formatting for each column or each row, but at least just avoid formatting floats so they can get printed in their entirety is a good first step.
